src/internal/ajax/types.ts

```diff
@@ -7,6 +7,10 @@ import { PartialObserver } from '../types';
  */
 export type AjaxDirection = 'upload' | 'download';
 
+export type ProgressEventType = 'loadstart' | 'progress' | 'load';
+
+export type AjaxResponseType = `${AjaxDirection}_${ProgressEventType}`;
```

Comment:
Template literal types require 4.1. I wasn't sure what the minimum supported TS version was. If needed I can swap this out for a regular union:

```suggestion
type AjaxResponseType =
  | 'upload_loadstart'
  | 'upload_progress'
  | 'upload_load'
  | 'download_loadstart'
  | 'download_progress'
  | 'download_load';
```

Reply:
4.2 is the minimum supported version, so template literal types are fine.

Reply:
You'll need to do an `npm run api_guardian:update`, too.